flake.nix

```
@@ -188,7 +188,7 @@
                 jdk17 = mkShell system { javaVersion = 17; };
                 jdk21 = mkShell system { javaVersion = 21; };
                 jdk22 = mkShell system { javaVersion = 22; };
-                extraChecks = mkShell system { extraChecks = true; javaVersion = 21; };
+                extraChecks = mkShell system { extraChecks = true; javaVersion = 22; };
```

```suggestion
                extraChecks = mkShell system { extraChecks = true; javaVersion = 23; };
```

22 is EOL, I would just go to 23

---

Done! A couple of lingering questions I had about JDK versions and `flake.nix` in general:

**Q1**: Should we remove the `jdk22` version from this file now, since it has reached EOL and looks like should not be used here anymore? 

**Q2**: Is there a way to test these changes to `flake.nix` locally? Or is running them in the context of Github actions the only way to verify this works?

---

**Q1**: I did that in a later commit in this PR

**Q2**: You can just copy-paste the command from Github Actions and it should reproduce exactly:
`nix develop --ignore-environment --keep MAVEN_OPTS --keep PATH --keep LANG .#extraChecks --command extra`